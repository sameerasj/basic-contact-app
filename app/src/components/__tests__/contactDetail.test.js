import { shallow } from "enzyme";
import { Link, BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ContactDetail from "../ContactDetail";

const name = "Test User";
const email = "testuser@mail.com";

const props = {
  location: {
    state: {
      contact: {
        name,
        email,
      }
    },
  },
};

it("should render ContactDetail", () => {
  const wrapper = shallow(<ContactDetail {...props}/>);
  expect(wrapper.find("img").props().src).toEqual("user.jpg");
  expect(wrapper.find(".header").text()).toEqual(name);
  expect(wrapper.find(".description").text()).toEqual(email);
});

it("should have Back button link", () => {
  const wrapper = shallow(<ContactDetail {...props}/>);
  const link = wrapper.find(Link);
  expect(link.props().to).toEqual("/");
  expect(link.children(0).text()).toEqual("Back to Contact List");
});

// Snapshot test
it("should match contactdetail snapshot", () => {
  const snap = renderer.create(
  <BrowserRouter>
  <ContactDetail {...props}/>
  </BrowserRouter>).toJSON();
  expect(snap).toMatchSnapshot();
})


