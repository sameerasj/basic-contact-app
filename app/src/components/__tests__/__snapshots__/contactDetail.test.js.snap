// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match contactdetail snapshot 1`] = `
<div
  className="main"
>
  <div
    className="ui card centered"
  >
    <div
      className="image"
    >
      <img
        alt="user"
        src="user.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Test User
      </div>
      <div
        className="description"
      >
        testuser@mail.com
      </div>
    </div>
  </div>
  <div
    className="center-div"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="ui button blue center"
      >
        Back to Contact List
      </button>
    </a>
  </div>
</div>
`;
